controllers/cinder_controller.go

```diff
@@ -95,317 +119,440 @@ func (r *CinderReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	instance.Status.APIEndpoints = cinderAPI.Status.APIEndpoints
+	instance.Status.ServiceIDs = cinderAPI.Status.ServiceIDs
+	instance.Status.CinderAPIReadyCount = cinderAPI.Status.ReadyCount
 
+	// TODO: These will not work without rabbit yet
 	// deploy cinder-scheduler
-	// Create or update the cinder-scheduler Deployment object
-	op, err = r.schedulerDeploymentCreateOrUpdate(instance)
+	cinderScheduler, op, err := r.schedulerDeploymentCreateOrUpdate(instance)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 	if op != controllerutil.OperationResultNone {
 		r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
 	}
 
+	// Mirror CinderScheduler status' ReadyCount to this parent CR
+	instance.Status.CinderSchedulerReadyCount = cinderScheduler.Status.ReadyCount
+
 	// deploy cinder-backup
-	op, err = r.backupDeploymentCreateOrUpdate(instance)
+	cinderBackup, op, err := r.backupDeploymentCreateOrUpdate(instance)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 	if op != controllerutil.OperationResultNone {
 		r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
 	}
 
-	// deploy cinder-volume services
-	// calc Spec.CinderVolumes hash to verify if any of the volume services changed
-	cinderVolumesHash, err := util.ObjectHash(instance.Spec.CinderVolumes)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("error volumes hash: %v", err)
-	}
-	r.Log.Info("VolumesHash: ", "Cinder-Volumes Hash:", cinderVolumesHash)
-
-	for _, volume := range instance.Spec.CinderVolumes {
-		// VolumeCustomConfigMap
-		volumeCMName := fmt.Sprintf("%s-%s-config-data-custom", instance.Name, volume.Name)
-		cm := []common.ConfigMap{
-			// CustomConfigMap
-			{
-				Name:      volumeCMName,
-				Namespace: instance.Namespace,
-				CMType:    common.CMTypeCustom,
-				Labels:    cmLabels,
-			},
-		}
-		err = common.EnsureConfigMaps(r, instance, cm, &envVars)
-		if err != nil {
-			return ctrl.Result{}, nil
-		}
+	// Mirror CinderBackup status' ReadyCount to this parent CR
+	instance.Status.CinderBackupReadyCount = cinderBackup.Status.ReadyCount
 
-		op, err = r.volumeDeploymentCreateOrUpdate(instance, &volume)
+	// TODO: This requires some sort of backend and rabbit, and will not work without them
+	// deploy cinder-volumes
+	for name, volume := range instance.Spec.CinderVolumes {
+		cinderVolume, op, err := r.volumeDeploymentCreateOrUpdate(instance, name, volume)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
 		if op != controllerutil.OperationResultNone {
-			r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", volume.Name, string(op)))
+			r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
 		}
+
+		// Mirror CinderVolume status' ReadyCount to this parent CR
+		// TODO: Somehow this status map can be nil here despite being initialized
```

Comment:
I tried adding that same pattern you linked from the Keystone operator to update the status, but I still seem to need these extra initializations or else it hits a null-pointer.  Could it perhaps be the opposite problem: 

1. Initialize empty map in the status
2. Save the status
3. Somehow status is read-back and map becomes nil again because it was empty during save?

Just spitballing, but this is definitely weird behavior.